students/alex_whitty/lesson4/trigrams.py

```diff
@@ -0,0 +1,83 @@
+import random
+import string
+#!/usr/bin/env python3
+def parse_file():
+    '''
+    Takes a hardcoded file name and returns a list of words
+    '''
+    # Setup a translator to remove punctuation
+    translator = str.maketrans('', '', string.punctuation)
+    # Setup an empty list to contain the full list of words in the text
+    full_list = []
+    with open('double_take.txt', 'r') as f:
+        for line in f:
+            # This will take each line, remove punctuation and then split it
+            # into a list of words and attach this list to the full_list
+            # note the use of "extend()" instead of "append()"
+            # TODO: Figure out how to remove carriage returns
+            full_list.extend(line.translate(translator).split(' '))
+    return full_list
+
+
+
+
+trigrams = {("When", "the"): ['Travelers'],
+            ("the", "Travelers"): ["from"],
+            ("Travelers", "from"): ["Outer"],
+            ("from", "Outer"): ["space"],
+            ("Outer", "Space"): ["dug"],
+            ("Space", "dug"): ["into"],
+            ("dug", "into"): ["the"],
+            ("into", "the"): ["pile"],
+            }
+
+
+
+def build_trigrams(words):
+
+    trigrams = {}
+
+    for i in range(len(words) - 2):
+
+
+        pair = tuple(words[i:i + 2])
+        follower = words[i + 2]
+
+        if pair not in trigrams:
+
+            trigrams[pair] = [follower]
+        else:
+            trigrams[pair].append(follower)
+
+
+    return trigrams
+
+# build up the dict here!
+
+def build_dict(words):
+    starter = random.choice(list(words.keys("from", "Outer")))
```

Comment:
For this one, you don't need to add ("from", "Outer"), the idea is that starter will use a random key from _words_